paper/paper.md

```diff
@@ -0,0 +1,201 @@
+
+## Constraints
+pyGeo also includes geometric constraints to prevent geometrically undesirable designs.
+The most commonly-used class of geometry constraints in pyGeo involves tracking one or more linear dimensions on the optimized object's surface.
+These constraints are created by specifying a single point, a line, or an array of points, along with a normal direction, then computing two line-surface intersection points.
+Some commonly used geometric constraints in shape optimization, such as thickness, area, and volume constraints (\autoref{fig:constraint}) can be computed using variations on this approach, which is computationally cheap and robust [@Brelje2020a].
+
+![Thickness and volume constraints demonstrated on an wing section [@Brelje2020a]. \label{fig:constraint}](constraints_3d.pdf)
+
+If a more complex geometry needs to be integrated into an optimized surface, pyGeo supports an alternative geometric constraint formulation based on arbitrary triangulated surfaces as described in [@Brelje2020a] and illustrated in \autoref{fig:trisurf}.
+
+![Triangulated surface constraint used to optimize an aeroshell around a complex geometry [@Brelje2020a].\label{fig:trisurf}](trisurfcon.pdf)
+
+
+# Parallelism
+pyGeo can optionally work under distributed memory parallelism using MPI, which is a requirement when interfacing with large-scale CFD applications.
+For example, the computational mesh may be partitioned and distributed among many processors by the CFD solver, and each processor may be aware of only its portion of the mesh.
+pyGeo can handle such scenarios by independently manipulating the geometry on each processor and aggregating the constraints across all processors when communicating with the optimizer.
+
+
+# Derivative Computation
+In addition to geometry manipulation and constraints, pyGeo can compute derivatives of these operations with respect to design variables.
+For the geometric deformation, pyGeo can compute the Jacobian
+\begin{equation*}
+\frac{\mathrm{d}X_s}{\mathrm{d}x} ,
+\end{equation*}
+where $X_s$ is the vector of surface mesh coordinates, and $x$ is the vector of geometric design variables.
+
+Similarly, pyGeo can compute the constraint Jacobian
+\begin{equation*}
+\frac{\mathrm{d}g}{\mathrm{d}x} ,
+\end{equation*}
+where $g$ is the vector of geometric constraints.
+
+For the FFD parameterization, these derivatives are computed using a combination of analytic methods [@Martins2021] and the complex-step method [@Martins2003a].
+For the interfaces to OpenVSP and ESP, the derivatives are computed with parallel finite differences.
+The CST derivatives are computed analytically.
+
+
+# Statement of Need
+Few open-source packages exist with comparable functionalities.
+To the authors' best knowledge, the only other open-source CFD-based optimization framework that contains a geometry parameterization is SU2 [@Economon2016a].
```

Comment:
Is "that contains a geometry parameterization" correct or missing something at the end?